compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -415,6 +415,15 @@ object Trees {
         qualifier.typeOpt.member(name).atSignature(Signature.NotAMethod, name)
       case _ =>
         super.denot
+
+    def nameSpan(using Context): Span =
```

Comment:
I tried this which did not need any attachments (I don't think its that important to do the point span for synthetic):
```scala
def nameSpan(using Context): Span =
  if span.exists then
    val point = span.point
    if name.toTermName == nme.ERROR then Span(point)
    else if qualifier.span.start > span.start then // right associative
      val realName = name.stripModuleClassSuffix.lastPart
      Span(span.start, span.start + realName.length, point)
    else
      Span(point, span.end, point)
  else span
```